hash_practice/exercises.py

```diff
@@ -22,8 +106,17 @@ def valid_sudoku(table):
         Each element can either be a ".", or a digit 1-9
         The same digit cannot appear twice or more in the same 
         row, column or 3x3 subgrid
-        Time Complexity: ?
-        Space Complexity: ?
-    """
-    pass
+        Time Complexity: Quadratic O(N^2) - nested for loop

```

Comment:
This would be true if we took in arbitrary nxn grids, but luckily since we only accept 3x3 grids, we can actually just say this is O(n)!